File: leibniz.lua
function leibniz_pow(i)
    return math.pow(-1, i) / (2 * i + 1)
end

function leibniz(n)
    r = 0
    for i = 0, n do
        r = r + leibniz_pow(i)
    end
    return 4 * r
end


print(leibniz(arg[1]))

File: leibniz.cs
using System;

class Leibniz
{

    public static double leibnizPow(double i)
    {
        return Math.Pow(-1, i) / (2 * i + 1);
    }

    public static double leibniz(int n)
    {
        double r = .0;
        for (int i = 0; i < n; i++)
        {
            r += leibnizPow(i);
        }
        return 4 * r;
    }

    public static void Main(string[] args)
    {
        int step = 0;
        if (int.TryParse(args[0], out step)) {
            double result = Leibniz.leibniz(step);
            Console.WriteLine(result);
        }
    }
}